### ISSUE TITLE
optimal dropout

### ISSUE THREAD

**Issue by LouisEchard** [LouisEchard/ML_MiniProjects#14]

**Comment by LouisEchard** [comment]
The key to understand the disharmony between Dropout and BN is the inconsistent
behaviors of neural variance during the switch of networks’ state. Considering one
87
neural response, when the state changes from train to test, Dropout would scale the
response by its Dropout retain ratio that actually changes the neural variance as
in learning, yet BN still maintains its statistical moving variance. This mismatch of
variance could lead to a numerical instability. As the signals go deeper, the numerical
deviation on the final predictions may amplify, which drops the system’s performance.
This scheme is known as “variance shift” for simplicity. Instead, without Dropout,
the real neural variances in inference would appear very closely to the moving ones
accumulated by BN, which is also preserved with a higher test accuracy.

Understanding the disharmony
between dropout and batch normalization by variance shift. Cornell University
Library, 2018. URL https://arxiv.org/abs/1801.05134.
